File: platforms/android/stato/src/main/java/com/facebook/stato/core/StatoStateUpdateListener.java
package com.facebook.stato.core;

public interface StatoStateUpdateListener {

  /**
   * Called when the state of the Stato client changes. Typical implementations will subscribe by
   * calling {@link com.facebook.stato.core.StatoClient#subscribeForUpdates()}, to start
   * receiving update events. Calling {@link com.facebook.stato.core.StatoClient#getState()}
   * will retrieve the updated state.
   */
  void onUpdate();
}

File: platforms/android/android-stato/src/main/kotlin/org/stato/core/StatoStateUpdateListener.kt
package org.stato.core

interface StatoStateUpdateListener {

  /**
   * Called when the state of the Stato client changes. Typical implementations will subscribe by
   * calling [org.stato.core.StatoClient.subscribeForUpdates], to start
   * receiving update events. Calling [org.stato.core.StatoClient.getState]
   * will retrieve the updated state.
   */
  fun onUpdate()
}
